omc3/tune_analysis/kick_file_modifiers.py

```diff
@@ -217,42 +247,134 @@ def get_ampdet_data(kickac_df, action_plane, tune_plane, corrected=False):
     data.columns = columns.keys()
 
     if data.isna().any().any():
-        LOG.warn(
+        # could be on purpose,
+        # for manually filtering kick points in one plane, but not in the other
+        LOG.warning(
             f"Amplitude Detuning data for Q{tune_plane} and J{action_plane} contains NaNs"
         )
-        data = data.dropna(axis=0)
-    return data
+        if dropna:
+            LOG.debug("NaN's are being dropped.")
+            data = data.dropna(axis=0)
+
+    return AmpDetData(
+        action_plane=action_plane,
+        tune_plane=tune_plane,
+        **{column: data[column] for column in data.columns}
+    )
 
 
 # Timed DataFrames -------------------------------------------------------------
 
 
-def get_timestamp_index(index):
+def get_timestamp_index(index: pd.Index) -> pd.Index:
     return pd.Index([i.timestamp() for i in index])
 
 
-def read_timed_dataframe(path):
+def read_timed_dataframe(path: Union[str, Path]) -> tfs.TfsDataFrame:
     df = tfs.read(path, index=get_time_col())
     df.index = pd.Index([CERNDatetime.from_cern_utc_string(i) for i in df.index], dtype=object)
     return df
 
 
-def write_timed_dataframe(path, df):
+def write_timed_dataframe(path: Union[str, Path], df: tfs.TfsDataFrame):
     df = df.copy()
     df.index = pd.Index([i.strftime(get_cern_time_format()) for i in df.index], dtype=str)
-    tfs.write(str(path), df, save_index=get_time_col())
+    tfs.write(path, df, save_index=get_time_col(), headerswidth=max(len(k) for k in df.headers.keys()))
```

Comment:
I'm thinking maybe we should have this be the default for `tfs-pandas` writing at some point if `headerswidth` is not given?

Reply:
possibly, yeah.